src/utils/LookmlDiagrammer/join-utils.ts

```diff
@@ -41,14 +41,14 @@ export function getPkJoinPathObj(
   pkTableRef: DiagramField[],
   pkFieldIndex: number
 ) {
-  return {
+  return ({
     viewName: join.name,
     fieldIndex: pkFieldIndex,
-    selector: pkTableRef[pkFieldIndex].name.replace('.', '-'),
+    selector: pkTableRef[pkFieldIndex]?.name?.replace('.', '-'),
```

Comment:
Do we have a good default value for `selector`? Not sure how `selector` is referenced later, but could leave it undefined, which could still lead to issues later. Looks like you set a default value on all the other values to be safe.

Reply:
Unfortunately not. I don't have a repro either. All I'm trying to do is fix the NPE. What happens after this I do not know